android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1784,6 +1787,48 @@ public Profile getCurrentProfile() {
         return Profile.fromWebContents(tab.getWebContents());
     }
 
+    public void closeExistingTabs(final boolean closeActiveTab, @NonNull final String baseUrl) {
+        final Tab activeTab = getActivityTab();
+        final TabModel tabModel = getCurrentTabModel();
+        if (activeTab != null
+                && activeTab.getUrl().getSpec().startsWith(baseUrl)
```

Comment:
comparing urls with startWith does not always work correct
like `brave://wallet` and `brave://walletsomething` would be a false positive
Most likely it should be implemented as searching for tabs by origin equality

Reply:
👍